Python/pystate.c

```diff
@@ -46,7 +46,12 @@ _PyRuntimeState_Init(_PyRuntimeState *runtime)
     _PyEval_Initialize(&runtime->ceval);
 
     runtime->gilstate.check_enabled = 1;
-    runtime->gilstate.autoTLSkey = -1;
+    /* A TSS key must be initialized with Py_tss_NEEDS_INIT
+       in accordance with the specification. */
+    {
+        Py_tss_t initial = Py_tss_NEEDS_INIT;
+        runtime->gilstate.autoTSSkey = initial;
+    }
```

Comment:
I have another idea instead of assignment by instant variable; provide a macro or inline function which ready the key (put into `Include/pythread.h`?).
```c
#ifndef Py_LIMITED_API
/* You *must not* use except for the case of initializing the key
   in the startup sequence! */
#define Py_tss_READY(key) do { (key)._is_initialized = 0; } while(0)
static inline void
PyThread_tss_ready(Py_tss_t *key) {
    key->_is_initialized = 0; 
}
#endif
```
In addition, calling `PyThread_tss_delete` gives to reset the key's initialization state, but it will shake the create/use/delete cycle. I don't like it.

Reply:
Since this is an internal use, I think it would be OK to just write `runtime->gilstate.autoTLSkey._is_initialized = 0`.

However, I don't consider this a merge blocker either way - the `Py_tss_NEEDS_INIT` version also works, and we can make "Provide a nicer runtime TSS structpre-initialization API" a separate RFE.

Reply:
I'd like to maintain the status quo to indicate that `Py_tss_t` is an opaque type, this is the one of the few chance.  In fact, except for testing, there is only another example `Module/_tracemalloc.c` in CPython repo. This is also the reason that first I chose the assignment by struct, but not the feature of initializing struct field (I'm not sure it will be required...).